src/main/java/org/elasticsearch/index/query/MatchAllQueryBuilder.java

```diff
@@ -100,15 +114,42 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
                 }
             }
         }
+    }
 
-        if (boost == 1.0f) {
-            return Queries.newMatchAllQuery();
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        this.boost = in.readFloat();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeFloat(this.boost);
+    }
+    
+    @Override
+    public int hashCode() {
+        int hash = super.hashCode();
+        hash = maybeHashcode(hash, boost);
+        return hash;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (getClass() != obj.getClass()) {
+            return false;
         }
 
-        //LUCENE 4 UPGRADE norms field is not supported anymore need to find another way or drop the functionality
-        //MatchAllDocsQuery query = new MatchAllDocsQuery(normsField);
-        MatchAllDocsQuery query = new MatchAllDocsQuery();
-        query.setBoost(boost);
-        return query;
+        MatchAllQueryBuilder other = (MatchAllQueryBuilder) obj;
+        return Objects.equals(boost, other.boost);
```

Comment:
Did this to be consistent with Lees PR here https://github.com/elastic/elasticsearch/pull/10405/files#diff-08a3154601eec8465b50ac68b45c4c48R523. 
Im happy to do this either way.